snooty/language_server.py

```diff
@@ -159,6 +159,11 @@ def on_update(self, prefix: List[str], page_id: FileId, page: types.Page) -> Non
     def on_delete(self, page_id: FileId) -> None:
         pass
 
+    def on_published_branches(
+        self, prefix: List[str], published_branches: SerializableType
```

Comment:
Should we pass `PublishedBranches` instead of the serialized form, and let the backend serialize things if it wants?